gateway/ln-gateway/src/lightning_manager.rs

```diff
@@ -31,25 +36,46 @@ impl LightningManager {
         Self {
             lightning_builder,
             state: Arc::new(RwLock::new(GatewayState::Initializing)),
+            htlc_task_group_or: Arc::new(RwLock::new(None)),
         }
     }
 
     pub async fn connect_route_htlcs(
         &self,
-        task_group: &mut TaskGroup,
+        task_group: TaskGroup,
     ) -> Result<(RouteHtlcStream, Arc<dyn ILnRpcClient>), LightningRpcError> {
+        // If the gateway is already connected, stop the current HTLC interception.
+        self.stop_route_htlcs().await;
```

Comment:
I don't think we did this before. Did you find that this was necessary?

Reply:
It's not strictly necessary, but I wanted the `LightningManager` to fully own its task group while providing an interface that contains as few foot-guns/idiosyncrasies as possible. `LightningManager::stop_route_htlcs` is idempotent (if it's not routing any HTLCs, it just returns without doing anything), and calling that here makes `LightningManager::connect_route_htlcs` idempotent as well (since that way we're not just tossing the old task group into the wind).